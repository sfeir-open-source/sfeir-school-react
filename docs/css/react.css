.reveal .slides section.center, .reveal .slides section .center {
  text-align: center; }

.reveal .slides section .big-text {
  font-size: 3em; }

.reveal .slides section.sfeir-basic-slide .center-big {
  text-align: center;
  font-size: 2.5em;
  color: black; }

.reveal .slides section.sfeir-basic-slide h1.center-big {
  font-size: 3em; }

.reveal .slides section.with-code pre.big-code:not(.literal), .reveal .slides section.with-code-dark pre.big-code:not(.literal) {
  position: relative;
  top: 0; }

.reveal .slides section img[alt*='full-width-screen'] {
  width: 100%;
  max-width: 100%;
  max-height: 100%; }

.reveal .slides section img[alt*='bottom'] {
  position: absolute;
  bottom: 0;
  left: 0; }

/**
Specifics
*/

/*# sourceMappingURL=react.css.map */