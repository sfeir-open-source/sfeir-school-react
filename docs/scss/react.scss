.reveal .slides section {
  &.center, .center {
    text-align: center;
  }
  .big-text {
    font-size: 3em;
  }

  &.sfeir-basic-slide {
    .center-big {
      text-align: center;
      font-size: 2.5em;
      color: black;
    }
    h1.center-big {
      font-size: 3em;
    }
  }

  &.with-code,
  &.with-code-dark {
    pre.big-code:not(.literal) {
      position: relative;
      top: 0;
    }
  }

  img[alt*='full-width-screen'] {
    width: 100%;
    max-width: 100%;
    max-height: 100%;
  }

  img[alt*='bottom'] {
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

/**
Specifics
*/

//@import './speaker.scss';

