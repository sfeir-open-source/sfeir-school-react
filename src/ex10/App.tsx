import React from "react";
import { Switch, Route, Redirect, RouteComponentProps } from "react-router-dom";

import { Header, HeaderActionItem } from "../solution/Header";
import { SearchableList } from "../solution/SearchableList";
import { Player } from "../solution/Player";
import { Loading } from "../solution/Loading";
import { Person } from "../solution/EditablePerson";

import { usePeople } from "./usePeople";

const ContextualList: React.FC = () => {
  const people = [];
  return <SearchableList people={people} />;
};

const ContextualPlayer: React.FC = () => {
  const people = [];
  return <Player people={people} />;
};

const ContextualPerson: React.FC<RouteComponentProps<{
  id: string;
}>> = ({ match }) => {
  const people = [];
  const person = people.find((p) => p.id === match.params.id);

  // TODO: call mutateAsync function from useMutation
  const onUpdate = (updated: Person) => {
    console.log("updated person:", updated);
    return Promise.resolve();
  }
  return <Person person={person} onUpdate={onUpdate} />;
};

export const App: React.FC = () => {
  // TODO: implement usePeople to load & store people
  const people = [];
  return (
    <>
      <Header>
        <HeaderActionItem to="/player" icon="view_carousel" />
        <HeaderActionItem to="/list" icon="view_module" />
      </Header>
      {people.length === 0 ? (
        <Loading />
      ) : (
        <Switch>
          <Route path="/list" component={ContextualList} />
          <Route path="/player" component={ContextualPlayer} />
          <Route path="/person/:id" component={ContextualPerson} />
          <Redirect to="/list" />
        </Switch>
      )}
    </>
  );
};
