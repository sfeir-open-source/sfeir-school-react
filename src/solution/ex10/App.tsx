import React from "react";
import { Switch, Route, Redirect, RouteComponentProps } from "react-router-dom";

import { Header, HeaderActionItem } from "../Header";
import { SearchableList } from "../SearchableList";
import { Player } from "../Player";
import { Loading } from "../Loading";
import { Person } from "../EditablePerson";
import { usePeople } from "./usePeople";

const ContextualList: React.FC = () => {
  const { people } = usePeople();
  return <SearchableList people={people} />;
};

const ContextualPlayer: React.FC = () => {
  const { people } = usePeople();
  return people && <Player people={people} />;
};

const ContextualPerson: React.FC<RouteComponentProps<{
  id: string;
}>> = ({ match }) => {
  const { people, updatePerson } = usePeople();
  const person = people.find((p) => p.id === match.params.id);
  return <Person person={person} onUpdate={updatePerson} />;
};

export const App: React.FC = () => {
  const { isLoading } = usePeople();

  return (
    <>
      <Header>
        <HeaderActionItem to="/player" icon="view_carousel" />
        <HeaderActionItem to="/list" icon="view_module" />
      </Header>
      {isLoading ? (
        <Loading />
      ) : (
        <Switch>
          <Route path="/list" component={ContextualList} />
          <Route path="/player" component={ContextualPlayer} />
          <Route path="/person/:id" component={ContextualPerson} />
          <Redirect to="/list" />
        </Switch>
      )}
    </>
  );
};
