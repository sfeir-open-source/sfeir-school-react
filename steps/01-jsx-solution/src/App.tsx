import { Header } from './components/Header';
import { Card } from './components/Card';

import './App.css';

const message = 'React @ SFEIR';

function App() {
  return (
    <>
      <Header />
      <Card title={message} />
    </>
  );
}

export default App;
