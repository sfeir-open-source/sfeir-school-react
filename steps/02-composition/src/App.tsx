import { Header } from './components/Header';
import { PersonCardModel } from './components/PersonCardModel';
import { PersonCard } from './components/PersonCard';

import './App.css';

type AppProps = {
  people: People;
};

function App({ people }: AppProps) {
  const randomPerson = people[Math.floor(Math.random() * people.length)];

  return (
    <>
      <Header />
      <main>
        <PersonCard person={randomPerson} />
        <PersonCardModel />
        <pre style={{ fontSize: 12 }}>{JSON.stringify(randomPerson, null, 2)}</pre>
      </main>
    </>
  );
}

export default App;
