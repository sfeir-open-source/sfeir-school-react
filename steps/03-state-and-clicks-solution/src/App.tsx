import { Header } from './components/Header';
import { PersonCard } from './components/PersonCard';

import './App.css';
import { Carousel } from './components/Carousel';

type AppProps = {
  people: People;
};

function App({ people }: AppProps) {
  const randomPerson = people[Math.floor(Math.random() * people.length)];

  return (
    <>
      <Header />
      <main>
        <PersonCard person={randomPerson} />
        <Carousel people={people} />
      </main>
    </>
  );
}

export default App;
