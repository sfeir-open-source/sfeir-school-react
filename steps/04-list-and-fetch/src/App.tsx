import { Header } from './components/Header';
import { PersonCard } from './components/PersonCard';
import { Carousel } from './components/Carousel';
import { people } from '../../api/db.json';

import './App.css';
import { PeopleGrid } from './components/PeopleGrid';

function App() {
  const randomPerson = people[Math.floor(Math.random() * people.length)];

  return (
    <>
      <Header />
      <main>
        <PersonCard person={randomPerson} />
        <Carousel people={people} />
        <PeopleGrid>{/* Put here all PersonCard */}</PeopleGrid>
      </main>
    </>
  );
}

export default App;
