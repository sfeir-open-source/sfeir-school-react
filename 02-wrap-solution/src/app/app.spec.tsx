import { render } from '@testing-library/react';
import App from './app';
describe('App', () => {
  let container: HTMLElement;
  let asFragment: any;

  beforeEach(() => {
    const { asFragment: snapshot, container: host } = render(<App />);
    asFragment = snapshot;
    container = host;
  });

  test('should display a snapshot of App', () => {
    expect(asFragment(<App />)).toMatchSnapshot();
  });

  test('should display in h1 tag the text "Hello Students', () => {
    const h1Element = container.querySelector('h1') as HTMLHeadingElement;
    expect(h1Element.textContent?.trim()).toEqual('Hello Students');
  });
});
