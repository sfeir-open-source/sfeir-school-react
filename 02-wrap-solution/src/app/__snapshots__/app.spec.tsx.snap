// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should display a snapshot of App 1`] = `
<DocumentFragment>
  <h1>
    Hello Students
  </h1>
</DocumentFragment>
`;
